create-meta/java/pom.xml

```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.cucumber</groupId>
+        <artifactId>cucumber-parent</artifactId>
+        <version>2.1.0</version>
+    </parent>
+
+    <artifactId>create-meta</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+    <name>Cucumber create meta</name>
+    <description>Produce the meta message for Cucumber Java</description>
+    <url>https://github.com/cucumber/create-meta-java</url>
+
+    <properties>
+        <project.Automatic-Module-Name>io.cucumber.createmeta</project.Automatic-Module-Name>
+    </properties>
+
+    <scm>
+        <connection>scm:git:git://github.com/cucumber/create-meta-java.git</connection>
+        <developerConnection>scm:git:git@github.com:cucumber/create-meta-java.git</developerConnection>
+        <url>git://github.com/cucumber/create-meta-java.git</url>
+    </scm>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest</artifactId>
+            <version>2.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.6.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+            <version>5.6.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>messages</artifactId>
```

Comment:
This will result in a rather annoying upgrade cycle. For every version of messages well have to release a new version of meta.

I'd suggest creating only the meta object here and copying the fields over in JVM.

Reply:
What if we moved `createMeta` to the `messages` module? That solves the upgrade cycle problem.

Reply:
The code looks great @vincent-psarga!

Reply:
It would break the upgrade cycle but would also make the messages lib heavier.

It is actually the same problem with Gherkin. And we don't solve that by pushing Gherkin into messages either.

Seperating the message and code that derives the information would be the best way.

Pushing down, worse then keeping it as is.

Reply:
> I'd suggest creating only the meta object here and copying the fields over to the message in JVM.

@mpkorstanje are you suggesting keeping the `createMeta` function in a separate module (`create-meta`), but decouple it from the `messages` module?

That works for me.

Reply:
I don't really see the point in decoupling to be honest. Can't we simply declare a minimum supported version of `messages` for the library.

To give an example, we release `messages` version 14.0 and `create-meta` version 1.0 (which relies on `messages` v14.0 minimum).

When we do a new release of `messages` but do not change the `Meta` message, we don't have anything to do in `create-meta` and no release needed.

When we do change the `Meta` message (adding a field or whatever), we'll need anyway to update the code of `create-meta` to handle those changes and populate the added field.

When we do a change in `create-meta` (for example extracting new CI information), we don't have to bump the minimum version of `messages` supported.

I'm pretty sure I'm wrong somehow, but I don't see why :D

Reply:
We'll have to fix the bit of code that removes version ranges when releasing the java modules.